### ISSUE TITLE
[Issue with Phone number component][<Maz ui Phone number>]

### ISSUE THREAD

**Issue by hadelnasr** [LouisMazel/maz-ui#1176]
## Describe the bug

Reference to the old hotfix for auto formatting issue here
https://github.com/LouisMazel/maz-ui/issues/1139

it produces another issue when setting the :auto-format="false" option, in the edit mode it's not displaying the phone number
## To Reproduce

add this logic in your vuejs code
`<MazPhoneNumberInput
      v-model="localPhoneNumber"
      :auto-format="false"
  />
const localPhoneNumber = props.phoneNumber;
`

where the props.phoneNumber equals: +201000000000 (egypt mobile number)

## Expected behavior

Should render this way
![image](https://github.com/user-attachments/assets/4c52fd69-ca89-4f71-a177-2894e3c582d2)

## Actual behavior
shown this way
![image](https://github.com/user-attachments/assets/45e4cd6d-fba5-46b0-a51c-356e93139b1a)

**Comment by LouisMazel** [comment]
This code is not correct

```ts
const localPhoneNumber = props.phoneNumber
```

`localPhoneNumber` is not reactive and can't be modify. 

Use it instead 

```ts
const localPhoneNumber = ref(props.phoneNumber)
```

**Comment by hadelnasr** [comment]
Dear @LouisMazel 

i tried adding your code and it's the same bug behavior
`const localPhoneNumber = ref(props.phoneNumber)`

**Comment by hadelnasr** [comment]
@LouisMazel any recommendations?

**Comment by LouisMazel** [comment]
Can you provide a reproduction link with stackblitz or any other ? 

I can't help you like this. 
All is fine with others users

**Comment by LouisMazel** [comment]
Working example: https://stackblitz.com/edit/maz-ui-vue3-phonenumber?file=src/App.vue

**Comment by hadelnasr** [comment]
> Working example: https://stackblitz.com/edit/maz-ui-vue3-phonenumber?file=src/App.vue

will you provide a quick example with a default phone number not empty string?

**Comment by LouisMazel** [comment]
Hi,

I released a beta version with the fix, can you try it and tell me if it's good?

The beta version is: `3.47.2-beta.14`

Install it: 

```
npm install maz-ui@3.47.2-beta.14
```
 
Thanks for your feedback :)

**Comment by hadelnasr** [comment]
@LouisMazel ok will give it a try and get back to you soon
thanks

**Comment by LouisMazel** [comment]
@Ziad-Ashraf-0 

Thank you a lot for the details. 

- It's now fixed

- For the second case (country with the same dialCode), it's because I deduce the country code based on the number. I change this so that it only happens when the phone number changes and not when the country code changes.

A new beta version is available if you want test it : `3.47.2-beta.18`

**Comment by Ziad-Ashraf-0** [comment]
@LouisMazel 

Thank you for the quick response—it's resolved now! 

However, I noticed some weird behavior with countries with a dialing code of `+1`. Some work fine, but others do not. For example, try a country like Bermuda, and input the example number `370-1234`. When you delete it, the "4" and "3" will get deleted as expected, but when you press backspace at "2," a strange behavior occurs—it duplicates the dialing code.
Also, if I try to input the example phone with the dialing code ( input 4413701 ), it duplicates the dialing code after I type the last "1".

I think it might be an issue with the `parsePhoneNumberFromString` function from `libphonenumber-js`.

Here is a link to reproduction indicating more examples
https://stackblitz.com/edit/maz-ui-vue3-phonenumber-kmzvzz?file=src%2FApp.vue

**Comment by hadelnasr** [comment]
@LouisMazel thanks for your quick response regarding the reported issue from @Ziad-Ashraf-0

**Comment by LouisMazel** [comment]
Released in [v3.48.0](https://github.com/LouisMazel/maz-ui/releases/tag/v3.48.0)
